beets/autotag/match.py

```diff
@@ -154,6 +149,10 @@ def track_index_changed(item: Item, track_info: TrackInfo) -> bool:
     return item.track not in (track_info.medium_index, track_info.index)
 
 
+track_length_grace = config["match"]["track_length_grace"].as_number()
+track_length_max = config["match"]["track_length_max"].as_number()
```

Comment:
I think it's common practice in our codebase to load configuration values at the point of use (unless there's a specific reason to do so earlier, I think there might be some plugins that do so on initialization).

Sometimes our tests depend on this (but apparently not here)?

Is there a specific reason for this change? Otherwise, I'd argue to keep it as is and move it back into `track_distance`.

Actually, this probably was a bottleneck of `track_distance`? I think this would be worth a code comment. Also, regarding the point above, maybe this could be moved to `assign_items` and the values passed to `track_distance`?

Reply:
Funnily, tests adjust global config so this may actually still work

> Also, regarding the point above, maybe this could be moved to assign_items and the values passed to track_distance?

But as you suggested I'd rather inject these values into the function so I will make this adjustment now.

> Actually, this probably was a bottleneck of track_distance?

Indeed. There's a fair bit of logic that runs to get these values and convert them to a number, so it becomes significant in larger albums. For example, I tested this trying to match my largest album with 136 to a candidate from Discogs that had 6666 tracks: in this case `track_distance` runs `136 * 6666 = 906,576` times.

Reply:
`track_distance` is called in 5 places across this module, so injection would have not been a clean solution. Instead, I added two functions that get these values and cached their calls.

Reply:
The new solution also isn't as clean as I'd like it to be, but on the other hand, if we don't depend on the config being evaluated at the time of the function call, it's probably good enough for now. We can revisit if that changes. Let's merge this as-is.